### PR TITLE
Unit test variantcall; fix/gather tests; fixes #55

### LINKED ISSUE #55: Re-structure tests
Create a general ```tests``` directory and move all the tests there. 

Could also study the possibility of mocking the DB to do CI in Travis, but let's put that last one in the wish list.

### PR DESCRIPTION
**Pull request by mariogiov**
Initial test that runs 2X data through Piper's merge_process_variantcall workflow. Also added some new utility functions I needed and updated the Charon tests to pass.
Oh, and finally collected the various test files to `ngi_pipeline.tests`.

### CODE REVIEW

File: ngi_pipeline/utils/pyutils.py [hunk 1]
```diff
@@ -10,3 +11,23 @@ def flatten(nested_list):
                 yield sub
         else:
             yield elt
+
+def update_dict(original_dict, updater_dict):
```
Comment (mariogiov) [inline ngi_pipeline/utils/pyutils.py]: I haven't tested this super extensively so caveat emptor.

Comment (mariogiov) [inline ngi_pipeline/utils/pyutils.py]: And it's my first shot at it so I may have missed something. It works for my config file.

File: milou_ngi_config.yaml [hunk 2]
```diff
@@ -18,12 +18,24 @@ environment:
     project_id: a2014205
     conda_env: ngi
     ngi_scripts_dir: /proj/a2014205/software/ngi_pipeline/scripts
+    flowcell_inbox: /proj/a2014205/archive
```
Comment (mariogiov) [inline milou_ngi_config.yaml]: Is this correct for milou?

File: milou_ngi_config.yaml [hunk 3]
```diff
@@ -52,3 +73,20 @@ slurm:
...
+        whole_genome_reseq:
+            test_project:
+                project_id: P0000
+                project_name: "Y.Mom_15_01"
+                bpa: "whole_genome_reseq"
+            customize_config:
+                analysis:
+                    top_dir: /proj/a2014205/nobackup/NGI/test_data/
+                charon:
+                    charon_base_url: http://charon-dev.scilifelab.se
```
Comment (mariogiov) [inline milou_ngi_config.yaml]: Hm. This won't work when we're trying to write back to charon-dev later. Ima think on this.

**Comment by mariogiov** [comment]
PR review fail I'm merging this

**Comment by guillermo-carrasco** [comment]
agh sorry :-( Ping me for the next one!

**Pull request merged**